### PR TITLE
Add FindCommand variants

### PR DESCRIPTION
**Pull request by macareonie**
FindCommand allows only for searching by name.

It would be beneficial to add flexibility to the find command by allowing keywords to find persons not only by name, but by phone and email as well.

Let's,
 * Add FindName/Phone/Email Commands with their respective Parser and Predicate classes.
 * Update Help window to include find command,
 * Update all relevant existing JUnit tests for FindNameCommand.
 * Add new JUnit tests for other FindCommand variants.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/FindEmailCommand.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.person.EmailContainsKeywordsPredicate;
+
+/**
+ * Finds and lists all persons in Tether whose email contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class FindEmailCommand extends FindCommand {
```
Comment (yashpola) [inline src/main/java/seedu/address/logic/commands/FindEmailCommand.java:14]: Good use of inheritance

File: src/main/java/seedu/address/ui/HelpWindow.java [hunk 2]
```diff
@@ -25,7 +25,8 @@ public class HelpWindow extends UiPart<Stage> {
             + "\n4. Delete an interview: delete_interview [interview index]"
             + "\n5. List applicants/interviewers: list_persons"
             + "\n6. List interviews: list_interviews"
-            + "\n7. Edit applicants/interviewers: edit [applicant/interviewer index] n/[newName]...";
+            + "\n7. Edit applicants/interviewers: edit [applicant/interviewer index] n/[newName]..."
+            + "\n8. Find person(s) by email/name/phone: find_[email/name/phone] [keyword1] [keyword2]...";
```
Comment (yashpola) [inline src/main/java/seedu/address/ui/HelpWindow.java:29]: Nice for remembering to edit help window for user convenience

File: src/main/java/seedu/address/model/person/PhoneContainsKeywordsPredicate.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    private final List<String> keywords;
+
+    public PhoneContainsKeywordsPredicate(List<String> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return keywords.stream()
+                .anyMatch(keyword -> StringUtil.containsWordIgnoreCase(person.getPhone().value, keyword));
```
Comment (yashpola) [inline src/main/java/seedu/address/model/person/PhoneContainsKeywordsPredicate.java:22]: May be possible to make phone matching more strict than contains. Interesting discussion on whether it makes sense for users to search phone numbers by substrings

File: src/test/java/seedu/address/logic/commands/FindEmailCommandTest.java [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.EmailContainsKeywordsPredicate;
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code FindCommand}.
+ */
+public class FindEmailCommandTest {
```
Comment (yashpola) [inline src/test/java/seedu/address/logic/commands/FindEmailCommandTest.java:26]: Good work for adding tests to cover both extensions of find command

**Review by yashpola — APPROVED** [review]

File: src/main/java/seedu/address/model/person/PhoneContainsKeywordsPredicate.java (hunk 3 above)
Comment (macareonie) [reply src/main/java/seedu/address/model/person/PhoneContainsKeywordsPredicate.java:22]: I referenced the original matching for find_name code in AB3. In this case, it allows for find_phone to search for multiple people at once. eg. "find_phone 123 456" will display people with phone number 123 and 456 (2 separate people)

Comment (macareonie) [reply src/main/java/seedu/address/model/person/PhoneContainsKeywordsPredicate.java:22]: As for phone matching, do you possibly be meaning to add the constraints in Phone class here as well? eg at least 3 digits etc.

**Review by headcube1 — APPROVED** [review]
LGTM

**Pull request merged**